src/sacn_parse_pack_error.rs

```diff
@@ -9,162 +9,55 @@
-            ///
-            PduInvalidLength(len: usize) {
-                description("Received PDU length is invalid"),
-                display("PDU Length {} is invalid", len)
-            }
-
-            /// Received PDU vector is invalid/unsupported by this library.
-            ///
-            /// # Arguments
-            /// vec: The vector parsed which is invalid / cannot be used.
-            ///
-            PduInvalidVector(vec: u32) {
-                description("Received PDU vector is invalid/unsupported by this library"),
-                display("Vector {:#x} not supported", vec)
-            }
-
-            /// Error parsing the received UUID.
-            ///
-            /// # Arguments
-            /// msg: A message providing further details (if any) as to why the uuid (used for CID) couldn't be parsed.
-            ///
-            UuidError(msg: String) {
-                description("Error parsing the received UUID"),
-                display("Error parsing the received UUID, msg: {}", msg)
-            }
-
-            /// Error parsing received UTF8 string.
-            ///
-            /// # Arguments
-            /// msg: A message providing further details (if any) as to why the string couldn't be parsed.
-            ///
-            Utf8Error(msg: String) {
-                description("Error parsing received UTF8 string"),
-                display("Error parsing received UTF8 string, msg: {}", msg)
-            }
-
-            /// Source name in packet was invalid, for example due to not being null terminated.
-            ///
-            /// # Arguments
-            /// msg: A message providing further details (if any) as to why the source name was invalid.
-            ///
-            SourceNameInvalid(msg: String) {
-                description("Attempted to parse invalid source name"),
-                display("Attempted to parse invalid source name, msg: {}", msg)
-            }
-        }
+    use thiserror::Error;
+
+    #[derive(Debug, Error)]
+    pub enum SacnParsePackError {
+        #[error("Error when parsing data into packet, msg: {0}")]
+        ParseInvalidData(String),
+
+        #[error("Attempted to parse a priority value that is outwith the allowed range of [0, 200], msg: {0}")]
+        ParseInvalidPriority(String),
+
+        #[error("Error when parsing page value, msg: {0}")]
+        ParseInvalidPage(String),
+
+        #[error("Attempted to parse a sync_addr value that is outwith the allowed range of [0, 63999], msg: {0}")]
+        ParseInvalidSyncAddr(String),
+
+        #[error("Attempted to parse a universe value that is outwith the allowed range of [1, 63999], msg: {0}")]
+        ParseInvalidUniverse(String),
+
+        #[error("Attempted to parse a packet with an invalid ordering of universes, msg: {0}")]
+        ParseInvalidUniverseOrder(String),
+
+        #[error("When packing a packet into a buffer invalid data encountered, msg: {0}")]
+        PackInvalidData(String),
```

Comment:
Rather than all the errors having 'msg: {0}' it would be better that they contain the invalid data if appropriate.

For example 'ParseInvalidUniverse(String)' should be 'ParseInvalidUniverse(u16)'